components/mediation/data-publishers/org.wso2.micro.integrator.observability/src/main/java/org/wso2/micro/integrator/observability/handler/metrichandler/prometheus/reporter/PrometheusReporter.java

```diff
@@ -0,0 +1,266 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.micro.integrator.observability.handler.metrichandler.prometheus.reporter;
+
+import io.prometheus.client.Counter;
+import io.prometheus.client.Gauge;
+import io.prometheus.client.Histogram;
+import io.prometheus.client.hotspot.DefaultExports;
+import org.wso2.config.mapper.ConfigParser;
+import org.wso2.micro.integrator.observability.handler.metrichandler.MetricReporter;
+import org.wso2.micro.integrator.observability.handler.util.MetricConstants;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Class for instrumenting Prometheus Metrics.
+ */
+public class PrometheusReporter implements MetricReporter {
+    private static Counter TOTAL_REQUESTS_RECEIVED_PROXY_SERVICE;
```

Comment:
Why do you need these to be static. If there variables are refered only within this class and since we only create single instance of Prometheus reporter you don't need this to be static

Reply:
Changed